### ISSUE TITLE
ETELEGRAM: 400 Bad Request: file is too big

### ISSUE THREAD

**Issue by knopki** [soberhacker/obsidian-telegram-sync#79]
Version: 1.1.1

When a message with a large video is sent to the bot, there is an error in the console. Neither the message nor the attachment is created.

```
Error: ETELEGRAM: 400 Bad Request: file is too big
    at eval (plugin:telegram-sync:60565:17)
From previous event:
    at Promise2.longStackTracesCaptureStackTrace [as _captureStackTrace] (plugin:telegram-sync:6737:27)
    at Promise2._then (plugin:telegram-sync:8516:23)
    at Promise2.then (plugin:telegram-sync:8430:25)
    at TelegramBotPolling._polling (plugin:telegram-sync:60219:48)
    at eval (plugin:telegram-sync:60262:58)
```

I wish:
1. Be able to download large files.
2. ~~If you can't download an attachment, at least save the message~~.

**Comment by knopki** [comment]
"Big file" is like 60 Mb.

**Comment by burtasov** [comment]
Hi! It will be resolved as soon as possible. 
There is an official limits for bots - 20 MB max file size possible to download. 
You can find approves here [bot api](https://core.telegram.org/bots/api) using search query "20 MB".
I have an idea how to skip this check, but need time to implement this.

**Comment by knopki** [comment]
Thank you so much!
But it is equally important to be able to handle any errors: try to save at least something (at least the text of the message), give feedback to the chat (the text of the error, for example).

**Comment by soberhacker** [comment]
Hi @knopki ,
Try 

> Thank you so much! But it is equally important to be able to handle any errors: try to save at least something (at least the text of the message), give feedback to the chat (the text of the error, for example).

Hi @knopki, try this new plugin release with error handling. Big Files support is still in process.

https://github.com/soberhacker/obsidian-telegram-sync/releases/tag/1.2.0
- add possibility to run plugin on a few desktops
- add displaying errors
- add forwardFrom to template parsing
- access allowed only for selected telegram users
- improve bot shutdown behavior when plugin is disabled and enabled again
- fix bug with non existed folder
- improve message deletion
- fix template variables names (context -> content)


Best Regards

**Comment by knopki** [comment]
Error is handled! Thank you!
![image](https://github.com/soberhacker/obsidian-telegram-sync/assets/891832/141125fb-bb8b-4095-8dd9-3a0795733e06)

Maybe it is worth to skip file with error in `handleFiles()` and save message anyway?

**Comment by soberhacker** [comment]
> Error is handled! Thank you!
> ![image](https://github.com/soberhacker/obsidian-telegram-sync/assets/891832/141125fb-bb8b-4095-8dd9-3a0795733e06)
> 
> Maybe it is worth to skip file with error in `handleFiles()` and save message anyway?

"...save message..." do you mean file caption?

**Comment by knopki** [comment]
> "...save message..." do you mean file caption?

⏳⏳⏳
Yes, I mean captions.

Sorry, I have no experience with the Telegram API and don't know specific terminology. But according to the code and Bot API documentation, I mean caption.

**Comment by soberhacker** [comment]
> > "...save message..." do you mean file caption?
> 
> ⏳⏳⏳ Yes, I mean captions.
> 
> Sorry, I have no experience with the Telegram API and don't know specific terminology. But according to the code and Bot API documentation, I mean caption.

Fixed in release 1.3.0
https://github.com/soberhacker/obsidian-telegram-sync/releases/tag/1.3.0

**Comment by knopki** [comment]
> Fixed in release 1.3.0

Works great! Thank you!

**Comment by soberhacker** [comment]
done
https://github.com/soberhacker/obsidian-telegram-sync/releases/tag/1.6.0
